### PR TITLE
Mimic Helm CRD install process by introducing wait for established

### PR DESCRIPTION
**Pull request by aiyengar2**
As part of the Helm CRD installation process shown [here](https://github.com/helm/helm/blob/43853ea772e779b617b011a8d4e69205636fa4f9/pkg/action/install.go#L160-L162), charts that both attempt to install CRDs and then install resources (e.g. `len(totalItems) > 0`) invalidate the cache and attempt to get the CRDs again to `Give time for the CRD to be recognized.` before continuing the Helm installation.

Because Monitoring V2 currently does not observe this invalidation and wait, flakey race conditions documented in issues like https://github.com/rancher/rancher/issues/32025 and https://github.com/rancher/rancher/issues/29171 can be caused. This commit introduces that invalidation process.

### CODE REVIEW

**Review by aiyengar2 — COMMENTED** [review]
This PR needs to be moved to the dev-v2.5 branch as the dev-v2.5-source branch is now deprecated.

The easiest way to do this would be to create a new local branch from the `dev-v2.5` branch, cherry-pick your commits from the branch this PR points to to that new branch, and then run `make charts` to generate the charts/ and assets/ as part of the new provisioning workflow.

More details on the new workflow can be found in https://github.com/rancher/charts/blob/dev-v2.5/packages/README.md#common-workflow. Feel free to find me on Slack / RocketChat or email me at arvind.iyengar@suse.com for any questions.

**Comment by aiyengar2** [comment]
Rebased to dev-v2.5

File: charts/rancher-monitoring/rancher-monitoring-crd/14.5.101+up14.5.0/templates/jobs.yaml [hunk 1]
```diff
@@ -45,7 +45,18 @@ spec:
...
+            echo "Waiting for CRDs to be recognized before finishing installation...";
+            
+            {{- range $path, $_ :=  (.Files.Glob  "crd-manifest/**.yaml") }}
+            {{- $apiGroup := get (get ($.Files.Get $path | fromYaml) "spec") "group" }}
+            rm -rf $HOME/.kube/cache/discovery/*/{{ $apiGroup }}
+            {{- end }}
+
+            {{- range $path, $_ :=  (.Files.Glob  "crd-manifest/**.yaml") }}
+            {{- $crd := get (get ($.Files.Get $path | fromYaml) "metadata") "name" }}
+            kubectl get crd {{ $crd }}
```
Comment (dramich) [inline charts/rancher-monitoring/rancher-monitoring-crd/14.5.101+up14.5.0/templates/jobs.yaml:58]: This call can/will fail, if we are attempting to duplicate what helm does it should be an error check and retry loop until the CRDs are available through the API 
https://github.com/helm/helm/blob/43853ea772e779b617b011a8d4e69205636fa4f9/pkg/kube/wait.go#L52:18

**Comment by aiyengar2** [comment]
Changed PR to dev-v2.6

**Comment by aiyengar2** [comment]
Probably resolves https://github.com/rancher/rancher/issues/32827 as well.

**Comment by aiyengar2** [comment]
PR is ready to be re-reviewed. 

I modified this PR according to @dramich's comment in https://github.com/rancher/charts/pull/1203#discussion_r638106337 and updated the initContainer jobs to also check whether the CRDs are established before proceeding.

One note: while Helm upstream checks whether either [`NamesAccepted=False || Established=True`](https://github.com/helm/helm/blob/43853ea772e779b617b011a8d4e69205636fa4f9/pkg/kube/wait.go#L307), my changes to the CRD chart strictly check if the CRDs are established before continuing since we cannot proceed with the installation if there is a naming conflict.

**Review by dramich — COMMENTED** [review]
@aiyengar2 Can we use a helper method here? This looks like a similar pattern in many places

**Comment by aiyengar2** [comment]
> @aiyengar2 Can we use a helper method here? This looks like a similar pattern in many places

@dramich addressed in latest commit. Good suggestion!

**Review by kinarashah — APPROVED** [review]
PR already reviewed by Dan. Went through the new helper changes for crd.established and looks good!

**Review by jiaqiluo — APPROVED** [review]
LGTM

**Pull request merged**